spec/controllers/char_attributes_controller_spec.rb

```diff
@@ -0,0 +1,186 @@
+require 'rails_helper'
+
+describe CharAttributesController, :type => :controller do
+  # have legit user, char_type and char_attribute mocks
+  before do
+
+    @user = double(User)
+    @char_type = double(CharType)
+    @char_attribute = double(CharAttribute)
+
+    allow(controller).to receive(:current_user).and_return(@user)
+    allow(controller).to receive(:authenticate_user!)
+
+  end
+
+  describe "#show" do
+
+    before do
+      find_char_type_stub
+      find_char_attribute_stub
+      get :show, valid_parameters
+    end
+
+    it "should have successful response" do
+      expect(response).to be_success
+    end
+
+  end
+
+  describe "#new" do
+
+    before do
+      find_char_type_stub
+    end
+
+    it "should make a new character attribute" do
+      allow(@char_type).to receive_message_chain(:char_attributes, :new)
+      get :new, :char_type_id => 46
+
+      expect(response).to be_success
+    end
+
+  end
+
+  describe "#edit" do
+
+    before do
+      find_char_type_stub
+      find_char_attribute_stub
+      get :edit, valid_parameters
+    end
+
+    it "should have successful response" do
+      expect(response).to be_success
+    end
+
+  end
+
+  describe "#create" do
+
+    before do
+      find_char_type_stub
+      allow(@char_type).to receive_message_chain(:char_attributes, 
+                                                 :build).and_return(@char_attribute)
+    end
+
+    context "character attribute is valid" do
+
+      before do
+        allow(@char_attribute).to receive(:save).and_return(true)
+        post :create, :char_type_id => 42, 
+                      :char_attribute => { :title => "Rogue", :value => "20" }
+      end
+
+      it "should create new attribute" do
+        expect(assigns(:char_attribute)).to eql(@char_attribute)
+      end
+
+      it "should redirect to character type page" do
+        expect(response).to redirect_to(char_type_path(@char_type))
+      end
+
```

Comment:
If you test redirecting status in one place, then you should test it in all places.